File: react-front/src/types/movie.ts
export interface Movie {
  id: number
  name: string
  url: string
  rating: number
  tags: string[]
}

export interface CreateMoviePayload {
  name: string
  url: string
  rating: number
  tags?: string[]
}

export function parseTagsInput(input: string): string[] {
  return input
    .split(',')
    .map((tag) => tag.trim())
    .filter(Boolean)
}

File: Net23Online/MovieMinimalApi/DbStuff/Movie.cs
namespace MovieMinimalApi.DbStuff
{
    public class Movie
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Url { get; set; } = string.Empty;
        public int Rating { get; set; }
        public ICollection<Tag> Tags { get; set; } = new List<Tag>();
    }
}
